eng/scripts/Automation-Sdk-Generate.ps1

```diff
@@ -0,0 +1,135 @@
+[CmdletBinding()]
+param(
+  [Parameter()]
+  [ValidateNotNullOrEmpty()]
+  [string] $InputJsonPath,
+
+  [Parameter()]
+  [string] $OutputJsonPath,
+
+  [Parameter()]
+  [string] $RepoRoot = "${PSScriptRoot}/../.."
+)
+
+$inputFileContent = Get-Content $InputJsonPath | ConvertFrom-Json
+[string[]] $inputFilePaths = $inputFileContent.changedFiles;
+$inputFilePaths += $inputFileContent.relatedReadmeMdFiles;
+$inputFilePaths = $inputFilePaths | Select-Object -Unique
+
+$changedFilePaths = $inputFilePaths -join "`n";
+Write-Host "List of changed swagger files and related readmes:`n$changedFilePaths`n"
+
+$autorestFilesPath = Get-ChildItem -Path "$RepoRoot/sdk"  -Filter autorest.md -Recurse | Resolve-Path -Relative
+
+Write-Host "Updating autorest.md files for all the changed swaggers."
+$sdksInfo = @{}
+$headSha = $inputFileContent.headSha
+$repoHttpsUrl = $inputFileContent.repoHttpsUrl
+foreach ($path in $autorestFilesPath) {
+  $fileContent = Get-Content $path
+  foreach ($inputFilePath in $inputFilePaths) {
+    $isUpdatedLines = $false
+    $escapedInputFilePath = [System.Text.RegularExpressions.Regex]::Escape($inputFilePath)
+    $regexForMatchingShaAndPath = "https:\/\/[^`"]*[\/][0-9a-f]{4,40}[\/]$escapedInputFilePath"
+    $updatedLines = foreach ($line in $fileContent) {
+      if ($line -match $regexForMatchingShaAndPath) {
+        $line -replace $regexForMatchingShaAndPath, "$repoHttpsUrl/blob/$headSha/$inputFilePath"
+
+        $isUpdatedLines = $true
+        $sdkpath = (get-item $path).Directory.Parent.FullName | Resolve-Path -Relative
+        if (!$sdksInfo.ContainsKey($sdkpath)) {
+          $specReadmePath = $inputFileContent.relatedReadmeMdFiles -match $inputFilePath
+          $sdksInfo.Add($sdkpath, $specReadmePath)
+        }
+      }
+      else {
+        $line
+      }
+    }
+    if ($isUpdatedLines) {
+      $updatedLines | Out-File -FilePath $path
+    }
+  }
+}
+
+Write-Host "Updated autorest.md files for all the changed swaggers. `n"
+
+$packages = @()
+foreach ($sdkPath in $sdksInfo.Keys) {
+  $packageName = Split-Path $sdkPath -Leaf
+  Write-Host "Generating code for " $packageName
+  $artifacts = @()
+  $hasBreakingChange = $null
+  $content = $null
+  $srcPath = Join-Path $sdkPath 'src'
+  dotnet msbuild /restore /t:GenerateCode $srcPath
+  if (!$LASTEXITCODE) {
+    $result = "succeeded"
+    Write-Host "Successfully generated code for" $packageName "`n"
+    
+    $csprojPath = Get-ChildItem $srcPath -Filter *.csproj -Recurse
```

Comment:
We are making an assumption that there is only one csproj found. If this ever returns more then one the line below it will fail in an unknown way. It might be worth checking that we only have one path.

Actually we could even consider just calling dotnet pack passing in the $srcPath as I think it will also find the project or give you an error about multiple projects if it finds more than one.

Reply:
Updated!